BSP/Simulator/Src/BSP_Timer.c

```diff
@@ -0,0 +1,76 @@
+#include "BSP_Timer.h"
+
+static void (*Timer1Handler)(int);
+static void (*Timer2Handler)(int);
+//had to pass some sort of parameter to the pointer or it doesn't work
+/** 
+
+ * @brief   Updates the time by reading the CSV file and calling relevant functions at the right time 
+
+ * @param   none
+
+ * @return  none
+
+ */  
+
+void BSP_Timer_Update()
+{
+   FILE *file = fopen("Timer.csv", "r");
+    int arr[4];
+
+    int current;
+
+    int reload;
+
+    int i=0;
+
+    int index=0;
+
+
+    while(i<2)
+    {
+        fscanf(file,"%d , %d",&current,&reload);

```

Comment:
If you're storing these values into `arr`, you don't need to have the two intermediate variables here. Also, try to use a more descriptive name than `arr`